tools/shared/src/main/scala/org/scalajs/core/tools/io/VirtualFiles.scala

```diff
@@ -132,7 +133,14 @@ trait VirtualSerializedScalaJSIRFile extends VirtualBinaryFile with VirtualScala
       ir.InfoSerializers.deserialize(stream)
     } catch {
       case e: IOException =>
-        throw new IOException(s"Failed to deserialize info of $path", e)
+        val pre = e match {
+          case e: IRVersionNotSupportedIOException =>
+            s"Requires Scala.js ${e.version} (or compatible version).\nCaused by: "
+
+          case _ => ""
+        }
+
+        throw new IOException(s"${pre}Failed to deserialize info of $path", e)
```

Comment:
I did not let it bubble up because the exception would not contain any reference to the file or library that was involved. For the format do you mean that this should throw again a `IRVersionNotSupportedIOException`.